src/Lykke.Service.PayInternal/Modules/CqrsModule.cs

```diff
@@ -0,0 +1,86 @@
+using Lykke.Service.PaySettlement.Contracts.Events;
+using Lykke.SettingsReader;
+using System.Collections.Generic;
+
+namespace Lykke.Service.PayInternal.Modules
+{
+    public class CqrsModule : Module
+    {
+        private readonly IReloadingManager<AppSettings> _appSettings;
+        public const string InternalBoundedContext = "lykkepay-internal";
+
+        public CqrsModule(IReloadingManager<AppSettings> appSettings)
+        {
+            _appSettings = appSettings;
+        }
+
+        protected override void Load(ContainerBuilder builder)
+        {
+            builder.RegisterType<SilentChaosKitty>()
+                    .As<IChaosKitty>()
+                    .SingleInstance();
+
+            builder.Register(context => new AutofacDependencyResolver(context)).As<IDependencyResolver>().SingleInstance();
+
+            RegisterComponents(builder);
+
+            const string settlementTransport = "Settlement";
+            var mainSettings = new RabbitMQ.Client.ConnectionFactory
+            {
+                Uri = _appSettings.CurrentValue.PaySettlementCqrs.ConnectionString
+            };
+
+            builder.Register(ctx =>
+            {
+                var logFactory = ctx.Resolve<ILogFactory>();
+
+                var settlementBroker = mainSettings.Endpoint.ToString();
+
+                var settlementMessagingEngine = new MessagingEngine(logFactory,
+                    new TransportResolver(new Dictionary<string, TransportInfo>
+                    {
+                        {
+                            settlementTransport,
+                            new TransportInfo(settlementBroker, mainSettings.UserName, mainSettings.Password, "None",
+                                _appSettings.CurrentValue.PaySettlementCqrs.Messaging)
+                        }
+                    }),
+                    new RabbitMqTransportFactory(logFactory));
+
+                return new CqrsEngine(logFactory,
+                    ctx.Resolve<IDependencyResolver>(),
+                    settlementMessagingEngine,
+                    new DefaultEndpointProvider(),
+                    true,
+                    Register.DefaultEndpointResolver(new RabbitMqConventionEndpointResolver(
+                        settlementTransport,
+                        _appSettings.CurrentValue.PaySettlementCqrs.SerializationFormat,
+                        environment: _appSettings.CurrentValue.PaySettlementCqrs.Environment)),
+
+                    Register.BoundedContext(InternalBoundedContext)
+                        .ListeningEvents(typeof(SettlementTransferToMarketQueuedEvent),
+                            typeof(SettlementTransferredToMerchantEvent), typeof(SettlementErrorEvent))
+                        .From(_appSettings.CurrentValue.PaySettlementCqrs.SettlementBoundedContext)
+                        .On(_appSettings.CurrentValue.PaySettlementCqrs.EventsRoute)
+                        .WithProjection(ctx.Resolve<SettlementProjection>(), 
```

Comment:
If I am not mistaken the commands should be used when there is a state change.

Reply:
There is no CQRS workflow in this project. Using saga with commands leads to surplus complexity in this case. I think will be better to add this feature to sage on moving this project to CQRS.